File: cpp/3x+1.cpp
#include <iostream>
using namespace std;

int main() {
    int i = 1;
    int n = 0;
    int count = 0;
    int max_count = 0;
    int max_i = 0;
    float start_time, end_time;

    start_time = clock();
    for (i = 1; i <= 1000000; i++) {
        n = i;
        count = 0;
        while (n != 1) {
            if (n % 2 == 0) {
                n = n / 2;
            } else {
                n = 3 * n + 1;
            }
            count++;
        }
        if (count > max_count) {
            max_count = count;
            max_i = i;
        }
    }
    end_time = clock();
    cout << max_i << " " << max_count << endl;
    cout << "Time: " << (end_time - start_time) / CLOCKS_PER_SEC << "s" << endl;
}

File: py/3x+1.py
import time
t1 = time.time()
max_count = 0
max_i = 0
for i in range(1000):
    count = 0
    n = i
    while n != 1:
        if n % 2 == 0:
            n = n // 2
        else:
            n = 3 * n + 1
        count += 1
    
    if count > max_count:
        max_count = count
        max_i = i

print(max_i, max_count)
print(time.time() - t1)